src/main/kotlin/org/anefdev/mixbuddy/util/PlaylistParser.kt

```diff
@@ -0,0 +1,168 @@
+package org.anefdev.mixbuddy.util
+
+import org.anefdev.mixbuddy.model.MusicPlaylist
+import org.anefdev.mixbuddy.model.MusicTrack
+import org.springframework.stereotype.Component
+import se.michaelthelin.spotify.model_objects.miscellaneous.AudioAnalysis
+import se.michaelthelin.spotify.model_objects.specification.PlaylistSimplified
+import se.michaelthelin.spotify.model_objects.specification.PlaylistTrack
+import se.michaelthelin.spotify.model_objects.specification.Track
+import kotlin.math.abs
+
+@Component
+class PlaylistParser {
+
+    fun getTrackListNoMetaData(playlist: List<PlaylistTrack>): List<MusicTrack> {
+        return playlist.stream().map { playlistTrack: PlaylistTrack ->
+            val musicTrack = MusicTrack()
+            val track = playlistTrack.track
+            musicTrack.id = track.id
+            musicTrack
+        }.toList()
+    }
+
+    fun getAllPlaylists(playlists: List<PlaylistSimplified>): List<MusicPlaylist> {
+        return playlists.stream().map { playlist: PlaylistSimplified ->
+            val musicPlaylist = MusicPlaylist()
```

Comment:
all `data class` properties should be passed to a constructor during instantiation, thats why they should be immutable and non-nullable with no default parameters if that is possible